confidant/models/credential.py

```diff
@@ -142,6 +160,40 @@ def _get_decrypted_credential_pairs(self):
         _credential_pairs = json.loads(_credential_pairs)
         return _credential_pairs
 
+    @property
+    def next_rotation_date(self):
+        """
+        Return when a credential needs to be rotated for security purposes.
+        """
+        if not self.requires_rotation or self.rotation_frequency is None:
```

Comment:
Short circuiting on `self.rotation_frequency is None` is risky in that if the env dict isn't present, we might not rotate.  I think it's worth doing this to be generic but I would expect other callers to ensure that the env var is present